src/libPersistence/BlockStorage.cpp

```diff
@@ -680,6 +688,29 @@ bool BlockStorage::GetStateRoot(bytes& data) {
   return true;
 }
 
+bool BlockStorage::GetLatestEpochStatesUpdated(uint64_t& epochNum) {
+  LOG_MARKER();
+
+  string epochNumStr;
+  {
+    shared_lock<shared_timed_mutex> g(m_mutexStateRoot);
+    epochNumStr = m_stateRootDB->Lookup(LATEST_EPOCH_STATES_UPDATED);
+  }
+
+  if (epochNumStr.empty()) {
+    LOG_GENERAL(INFO, "No Latest Epoch State Updated get");
+    return false;
+  }
+
+  try {
+    epochNum = stoul(epochNumStr);
```

Comment:
`stoull` should be used here instead of `stoul`, because the epochNum is `uint64_t`.
And, if possible, please also change the other wrong usage inside the functions: `GetDiagnosticDataNodes` and `GetDiagnosticDataCoinbase`.
Thanks!

Reply:
Fixed